### PR TITLE
[Beam-4115] Enable Go SDK vanity import path

### PR DESCRIPTION
**Pull request by damondouglas**
The purpose of this PR is for review purposes to make progress towards a Go SDK vanity import path.

This would enable:

```go
import "beam.apache.org/sdks/go/pkg/beam"
```

instead of:

```go
import "github.com/apache/beam/sdks/go/pkg/beam"
```

I would like to request @lostluck to review this PR.

My questions/comments with this PR are:
- I wasn't sure if it made sense to break this into smaller commits.  Should I retract this PR and start over?
- My experience with htaccess files are limited and I didn't have a good way of testing the changes
- It was mentioned in slack that the timing of this PR should wait until BEAM-9615 - Beam Schema Support is complete
- Both integration and regression tests passed, yet I did not have success running the [sdks/go/test/run_integration_tests.sh](sdks/go/test/run_integration_tests.sh) on my machine and received the error: `gcloud crashed (TypeError): a bytes-like object is required, not 'str'`
- Do the amount of changes associated with this PR, does it warrant updating `CHANGES.md`?

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/go/go-import.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<!--
```
Comment (damondouglas) [inline sdks/go/go-import.html:1]: This file needs to move into the website/www/site/static/go/sdk directory.

**Comment by lostluck** [comment]
Wowee! Big change :D

I don't think it's possible to break it up though? Vanity URLs are tricky in that respect and I wrote that original doc before Modules were available in earnest.

I do agree that it would warrant a mention in the CHANGES file.

The big trick here will be making sure that the jenkins tests also all pass before merging. No rush, as we aren't likely going to merge this in before the 2.26 cut (november 5th), to make sure that we can get it properly stable before the next cut.

Another thing to consider is whether to also make the module + v2 switch happen at the same time, since that's what the git repo tags will be saying anyway.

**Comment by damondouglas** [comment]
After I did this PR and saw the failing tests I cringed 🤣.  Thank you for still accepting me working on this.  I've already learned a lot about how we set up Gradle tasks and Jenkins.  Here's my plan:

1. Test the htaccess changes using a domain I control. It still won't be ideal since the actual beam.apache.org domain won't be tested but it's better than nothing.
2. Go deeper into the Jenkins setup to see why it wasn't finding certain dependencies.
3.  Investigate the module + v2 switch and the needs for this to be accomplished.

Oi vey

**Comment by lostluck** [comment]
Most of it has to do with the vendoring. It ends up being a bit poorly recursive in some cases. It might be better to do the switch to Go Modules first, which should then make this easier, since then we can keep Gradle from meddling with the go dependencies.

eg. sometimes it goes too deep:
.gogradle/project_gopath/src/github.com/apache/beam/sdks/java/boot/vendor/github.com/apache/beam/sdks/go/pkg/beam/util/grpcx/hook.go:23:2: cannot find package "beam.apache.org/sdks/go/pkg/beam/core/util/hooks" in any of:
17:34:00 	/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Go_Commit/src/sdks/java/container/.gogradle/project_gopath/src/github.com/apache/beam/sdks/java/boot/vendor/beam.apache.org/sdks/go/pkg/beam/core/util/hooks (vendor tree)
17:34:00 	/usr/lib/go-1.12/src/beam.apache.org/sdks/go/pkg/beam/core/util/hooks (from $GOROOT)
17:34:00 	/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Go_Commit/src/sdks/java/container/.gogradle/project_gopath/src/beam.apache.org/sdks/go/pkg/beam/core/util/hooks (from $GOPATH)

**Comment by damondouglas** [comment]
I want to communicate my progress even if I'm still not successful:
1.  I've been able to get `./gradlew goPreCommit` to pass on my machine only by inserting `t.Skip()` on [TestSym2Addr](https://github.com/apache/beam/blob/master/sdks/go/pkg/beam/core/util/symtab/symtab_test.go#L73).  I received the error message `symtab_test.go:113: dyld: malformed mach-o image: segment __DWARF has vmsize < filesize`.
2.  I setup the changes locally changing the package prefixes to `beam.ngrok.io`.  While I've successfully ran the website locally, tunneling using `beam.ngrok.io` and forwarding to the `go-import.html` file, I have not been successful with `go get beam.ngrok.io/sdks/go/pkg/beam`.

This is my plan.  Launching the site locally, involving the .htaccess file takes too long to realize the effect of changes between this file and the go-import.html file.  I am going to establish success with a very small separate hello world go project using the beam.ngrok.io domain and a simple nginx service.  This could serve as a baseline to then apply the insights into this project.

**Pull request closed without merging**